backend/api/receipts/download.py

```diff
@@ -0,0 +1,64 @@
+from django.http import FileResponse, Http404
+from backend.models import Receipt, ReceiptDownloadToken
+from django.shortcuts import get_object_or_404, render
+from django.urls import reverse
+from django.http import HttpResponse, JsonResponse
+
+
+def download_receipt(request, token):
+    """
+    Downloads a receipt file based on the provided token.
+
+    Args:
+        request (HttpRequest): The HTTP request object.
+        token (str): The token used to authenticate the download.
+
+    Returns:
+        HttpResponse: The HTTP response containing the downloaded receipt file.
+    """
+    try:
+        download_token = ReceiptDownloadToken.objects.get(token=token)
+    except ReceiptDownloadToken.DoesNotExist:
+        return HttpResponse("Download link has been used", status=404)
+
+    # if download_token.is_used():
+    #     return HttpResponse("Download link has been used", status=410)  # 410 Gone
+    if download_token.user != request.user:
+        return HttpResponse("Invalid user", status=403)  # 403 Forbidden
```

Comment:
Great! Instead of saying "Invalid user" could just say "Forbidden", same with the comment above this, just so the user doesn't get any hints. But great idea with this logic to make sure that users can't access anyones receipt!